java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -910,7 +910,9 @@ private void readFromJsonIntoVector(Field field, FieldVector vector) throws IOEx
       }
 
       int nullCount = 0;
-      if (type instanceof ArrowType.Null) {
+      if (type instanceof ArrowType.RunEndEncoded) {
```

Comment:
We could fold the union check below into this one?

Reply:
vectorBuffers is empty for RunEndEncoded, so vectorBuffers.get(0) fails.

Reply:
What I mean is 

```
if (run end || union) {
  nullCount = 0;
} else if (null) {
  nullCount = valueCount;
} else {
  // compute null count since it is not run end, union, or null
}
```

Reply:
I got you, you mean like 
``` Java
int nullCount;
if (type instanceof ArrowType.RunEndEncoded || type instanceof Union) {
  nullCount = 0;
} else if (type instanceof ArrowType.Null) {
  nullCount = valueCount;
} else {
  nullCount = BitVectorHelper.getNullCount(vectorBuffers.get(0), valueCount);
}
```

Reply:
Yup!